gluestick/singer.py

```diff
@@ -323,18 +325,25 @@ def to_singer(
         The output file name.
     allow_objects: boolean
         Allow or not objects to the parsed, if false defaults types to str.
+    keep_null_fields: boolean
+        Flag to keep all null fields
+    catalog_stream: str
+        Name of the stream in the catalog to be used to generate the schema if USE_CATALOG_SCHEMA is set as true
+        If this is not set it will use stream parameter to generate the catalog
 
     """
     catalog_schema = os.environ.get("USE_CATALOG_SCHEMA", "false").lower() == "true"
     include_all_unified_fields = os.environ.get("INCLUDE_ALL_UNIFIED_FIELDS", "false").lower() == "true" and unified_model is not None
 
-    if allow_objects and not (catalog_schema or include_all_unified_fields):
+    # drop columns with all null values except when we want to keep null fields
+    if allow_objects and not (catalog_schema or include_all_unified_fields or keep_null_fields):
         df = df.dropna(how="all", axis=1)
 
-    if catalog_schema:
+    if catalog_schema or catalog_stream:
         # it'll allow_objects but keeping all columns
         allow_objects = True
         # get schema from catalog
+        stream = catalog_stream or stream
```

Comment:
@keyn4  won't this change not just the lookup but also the export stream name? Meaning if I did
```python
to_singer("contacts", catalog_stream="customers')
```
it would result in the output being called `customers` instead of `contacts`?